File: common/src/main/java/com/hartwig/actin/treatment/input/single/TwoIntegers.java
package com.hartwig.actin.treatment.input.single;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class TwoIntegers {

    public abstract int integer1();

    public abstract int integer2();
}

File: common/src/main/java/com/hartwig/actin/treatment/input/single/TwoIntegers.kt
package com.hartwig.actin.treatment.input.single

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class TwoIntegers() {
    abstract fun integer1(): Int
    abstract fun integer2(): Int
}
